andela/ah-django-unchained
# #162948843 Implement Rating of Articles

**What does this PR do?**

Implements rating of articles where ratings are integers from 1 to 5.

**Description of tasks to be completed**

- create article rating model
- create an article rating serializer class
- add view for viewing an article's average ratings
- add view for posting a rating to an article
- refactor tests

**How should this be manually tested?**

- `git clone https://github.com/andela/ah-django-unchained.git`
- `git checkout ft-article-rating-162948843`
- `source .env`
- `python migrate`
- run tests `python manage.py test`

On Postman, register 2 or more users via the endpoint `http://127.0.0.1:8000/api/users/`.

Create an article while signed in via the endpoint `http://127.0.0.1:8000/api/articles/` using one of the users details.

To rate the article:
Make a POST request on `http://127.0.0.1:8000/api/articles/<slug>/rate/`. Ensure to copy the token of one of the other user and add it as an authorization header in the request.
The request body should look like this;
`
{
	"rate": 2
}
`
The response is a message that the article has been successfully rated.

To view the ratings of a specific article:
Make a GET request on `http://127.0.0.1:8000/api/articles/rate/<slug>/`.
The average ratings of that article should be displayed in the response.

**Any background context you want to provide?**

A user cannot rate their own article

**Relevant Pivotal Tracker Story**

[#162948843](https://www.pivotaltracker.com/story/show/162948843)

**Screenshots**

Rate article
<img width="976" alt="screenshot 2019-01-28 at 09 50 16" src="https://user-images.githubusercontent.com/32644295/51819187-91779f80-22e2-11e9-85f8-c066aceaa462.png">




View article ratings
<img width="981" alt="screenshot 2019-01-25 at 09 17 51" src="https://user-images.githubusercontent.com/32644295/51728691-2a5bb000-2082-11e9-8be3-5dfb3910ea15.png">

Give a rating of more than 5
<img width="963" alt="screenshot 2019-01-27 at 12 23 33" src="https://user-images.githubusercontent.com/32644295/51799119-7d726600-222e-11e9-9ae0-e09d32a4c443.png">
